storage/store.go

```diff
@@ -1701,6 +1707,42 @@ func (s *Store) MergeRange(subsumingRng *Replica, updatedEndKey roachpb.RKey, su
 	return subsumingRng.setDesc(&copy)
 }
 
+// If the subsuming replica has the range lease, then we update its timestamp
+// cache with the entries from the subsumed. If it's not the lease holder,
+// then the timestamp cache doesn't matter (in fact it should be empty, to
+// save memory).
+func (s *Store) maybeMergeTimestampCaches(
+	subsumingRng *Replica,
+	subsumedRng *Replica,
+) error {
+	subsumingRng.mu.Lock()
+	defer subsumingRng.mu.Unlock()
+	subsumingLease := subsumingRng.mu.state.Lease
+
+	subsumedRng.mu.Lock()
+	defer subsumedRng.mu.Unlock()
+	subsumedLease := subsumedRng.mu.state.Lease
+
+	// TODO(andrei): I think checking the subsumed leases like this is incorrect,
+	// since the stores of the different replicas of subsuming can have different
+	// state for subsumed? Unless the subsumed range has been quiesced somehow.
```

Comment:
The subsumed race will need to be frozen somehow. Merge support is currently very incomplete: #2433.